components/esp_lvgl_port/esp_lvgl_port.c

```diff
@@ -784,6 +916,323 @@ static void lvgl_port_btn_up_handler(void *arg, void *arg2)
 }
 #endif
 
+#if __has_include ("usb/hid_host.h")
+static lvgl_port_usb_hid_ctx_t *lvgl_port_hid_init(void)
+{
+    esp_err_t ret;
+
+    /* USB HID is already initialized */
+    if (lvgl_port_ctx.hid_ctx.task) {
+        return &lvgl_port_ctx.hid_ctx;
+    }
+
+    /* USB HID host driver config */
+    const hid_host_driver_config_t hid_host_driver_config = {
+        .create_background_task = true,
```

Comment:
Now we have 3 tasks to handle HID events:
1. USB Host library task (must be present in the application)
2. HID driver task
3. LVGL/HID glue task

I hope that 2 and 3 can be combined. Could you please help with this @roma-jam?

Reply:
Hey @espzav, @tore-espressif 
Yes, there is an option to decrease the task creation and merging tasks 2 and 3 with the following steps: 
1. Use the `create_background_task = false` in config during the `hid_host_install()`
2. After this, we need to call `esp_err_t hid_host_handle_events(uint32_t timeout_ms)` from some task, I assume that could be the LVGL/HID glue task. 

Hope that can help.
If there is anything else, I am ready to help, just poke me.

Reply:
What do you folks think of not creating the HID task in LVGL port layer? Given the purpose of LVGL port component, it's a bit unexpected that it would handle initialization of HID. Could we either 1) let the application initialize the HID and register some callback of BSP, or 2) move this functionality into a separate optional component, esp_lvgl_port_hid, which would encapsulate all the logic related to HID bindings to LVGL.

Reply:
@roma-jam I tried `create_background_task = false` and call `hid_host_handle_events` from  LVGL/HID glue task. After that, the function `hid_class_request_set_idle` crashes:
```
E (5897) hid-host: Control Transfer Timeout
E (5897) hid-host: hid_control_transfer(801): Unable to HALT EP
ESP_ERROR_CHECK failed: esp_err_t 0x102 (ESP_ERR_INVALID_ARG) at 0x42009830
```

Reply:
@espzav Thanks for the report. Working on it.